llama-index-core/tests/workflow/test_streaming.py

```diff
@@ -0,0 +1,53 @@
+import asyncio
+
+import pytest
+
+from llama_index.llms.openai import OpenAI
+
+from llama_index.core.llms import ChatMessage, MessageRole
+from llama_index.core.workflow.context import Context
+from llama_index.core.workflow.decorators import step
+from llama_index.core.workflow.events import Event, StartEvent, StopEvent
+from llama_index.core.workflow.workflow import Workflow
+from llama_index.core.workflow.errors import WorkflowRuntimeError
+
+
+class StreamingWorkflow(Workflow):
+    @step
+    async def chat(self, ctx: Context, ev: StartEvent) -> StopEvent:
+        llm = OpenAI()
+        messages = [
+            ChatMessage(
+                role=MessageRole.SYSTEM,
+                content="You are a pirate with a colorful personality",
+            ),
+            ChatMessage(role=MessageRole.USER, content="Who is Paul Graham?"),
+        ]
+        gen = await llm.astream_chat(messages)
+        async for resp in gen:
+            ctx.session.write_stream_event(Event(msg=resp.delta))
```

Comment:
session was already exposed because of `send_event` so I thought I would keep the "family feeling" but honestly I'm not sure, the first version was indeed `ctx.write_stream_event`...

Reply:
ohhh I think I missed that send_event wasn't on the context, but was on the session 👀

Reply:
I think we have two options:
- We use `Context` as a facade for everything session-related, until it's time to make session part of the public API
- We start exposing users to the concept of session very gently with only two fairly advanced use cases (manual sending events and streaming)

I would go with option n.2 (the one in this PR) if we are positive the session will be eventually made more prominent

Reply:
Yea i agree, option 2 makes sense here